### PR TITLE
Trying to rephrase complex sentences to make them easier to read.

### PR DESCRIPTION
**Pull request by Mbodin**
**Kind:** documentation.

I just noticed some typos and difficult-to-parse sentences.

### CODE REVIEW

File: doc/sphinx/language/extensions/evars.rst [hunk 1]
```diff
@@ -44,12 +44,14 @@ existential variable is represented by “?” followed by an identifier.
...
+form :n:`{ {*; @ident := @term} }`, is appended to its name, indicating
 how the variables of its defining context are instantiated.
-The variables of the context of the existential variables which are
-instantiated by themselves are not written, unless the :flag:`Printing Existential Instances` flag
-is on (see Section :ref:`explicit-display-existentials`), and this is why an
-existential variable used in the same context as its context of definition is written with no instance.
+The variables of the existential variables's context that are
+instantiated by themselves are hidden:
+this is why an existential variable used in the same context than its
+context of definition is by default written with no instance.
```
Comment (jfehrle) [inline doc/sphinx/language/extensions/evars.rst:52]:
```suggestion
context of definition is written with no instance.
```

File: doc/sphinx/language/extensions/evars.rst [hunk 2]
```diff
@@ -13,7 +13,7 @@ Existential variables
    | ?[ ?@ident ]
    | ?@ident {? @%{ {+; @ident := @term } %} }
 
-|Coq| terms can include existential variables which represents unknown
+|Coq| terms can include existential variables which represent unknown
```
Comment (jfehrle) [inline doc/sphinx/language/extensions/evars.rst:16]:
```suggestion
|Coq| terms can include existential variables that represent unknown
```

File: doc/sphinx/language/extensions/evars.rst [hunk 3]
```diff
@@ -13,7 +13,7 @@ Existential variables
...
 subterms to eventually be replaced by actual subterms.
```
Comment (jfehrle) [inline doc/sphinx/language/extensions/evars.rst:17]:
```suggestion
subterms that are eventually replaced with actual subterms.
```

**Review by jfehrle — COMMENTED** [review]
Thanks for the PR.  I don't know this feature very well, so my suggestions are copy editing only.  They may have inadvertently changed the meaning a little.

File: doc/sphinx/language/extensions/evars.rst [hunk 4]
```diff
@@ -44,12 +44,14 @@ existential variable is represented by “?” followed by an identifier.
 
 In the general case, when an existential variable :n:`?@ident` appears
 outside of its context of definition, its instance, written under the
-form :n:`{ {*; @ident := @term} }` is appending to its name, indicating
+form :n:`{ {*; @ident := @term} }`, is appended to its name, indicating
 how the variables of its defining context are instantiated.
-The variables of the context of the existential variables which are
-instantiated by themselves are not written, unless the :flag:`Printing Existential Instances` flag
```
Comment (corwin-of-amber) [inline doc/sphinx/language/extensions/evars.rst:50]: Is the part about `Printing Existential Instances` still relevant? I agree that the text is a bit cryptic here. Perhaps just a "See also" is in order.

**Comment by Mbodin** [comment]
Thanks for the suggestions! I’m on my way to amend my changes.

File: doc/sphinx/language/extensions/evars.rst [hunk 5]
```diff
@@ -44,12 +44,14 @@ existential variable is represented by “?” followed by an identifier.
 
 In the general case, when an existential variable :n:`?@ident` appears
 outside of its context of definition, its instance, written under the
```
[the remark this answers is not in the document]
Comment (Mbodin) [reply doc/sphinx/language/extensions/evars.rst:46]: Thanks for this! I did hesitate on the “under”, but wasn’t sure.

File: doc/sphinx/language/extensions/evars.rst (hunk 4 above)
Comment (Mbodin) [reply doc/sphinx/language/extensions/evars.rst:50]: It is relevant, but I agree that the place of the example is wrong: it should be in the part describing the `Printing Existential Instances` flag. I’ve just pushed a commit with the change.

Comment (Mbodin) [reply doc/sphinx/language/extensions/evars.rst:50]: About the relevance:
```coq
Check (fun x y => _) 0 1. (* (fun x y : nat => ?y) 0 1 : ?T@{x:=0; y:=1} *)
Set Printing Existential Instances.
Check (fun x y => _) 0 1. (* (fun x y : nat => ?y@{x:=x; y:=y}) 0 1 : ?T@{x:=0; y:=1} *)
```

File: doc/sphinx/language/extensions/evars.rst [hunk 6]
```diff
@@ -44,12 +44,14 @@ existential variable is represented by “?” followed by an identifier.
...
-form :n:`{ {*; @ident := @term} }` is appending to its name, indicating
+form :n:`{ {*; @ident := @term} }`, is appended to its name, indicating
 how the variables of its defining context are instantiated.
-The variables of the context of the existential variables which are
-instantiated by themselves are not written, unless the :flag:`Printing Existential Instances` flag
-is on (see Section :ref:`explicit-display-existentials`), and this is why an
-existential variable used in the same context as its context of definition is written with no instance.
+The variables of the existential variables's context that are
```
[the remark this answers is not in the document]
Comment (jfehrle) [reply doc/sphinx/language/extensions/evars.rst:49]: This should be the possessive form of the singular `variable`, right?  That would be `variable's`.  The possessive of the plural `variables` is `variables'`.  Yes, it looks strange, but it is correct.

See [here](https://www.thepunctuationguide.com/apostrophe.html#:~:text=The%20general%20rule%20is%20that,ends%20in%20s%20or%20not.&text=The%20possessive%20of%20a%20plural,a%20letter%20other%20than%20s.)

File: doc/sphinx/language/extensions/evars.rst [hunk 7]
```diff
@@ -13,8 +13,8 @@ Existential variables
    | ?[ ?@ident ]
    | ?@ident {? @%{ {+; @ident := @term } %} }
 
-|Coq| terms can include existential variables which represents unknown
-subterms to eventually be replaced by actual subterms.
+|Coq| terms can include existential variables that represent unknown
+subterms that are eventually replaced with actual subterms.
 
 Existential variables are generated in place of unsolvable implicit
```
Comment (jfehrle) [inline doc/sphinx/language/extensions/evars.rst:19]:
```suggestion
Existential variables are generated in place of unsolved implicit
```

Maybe "unknown" is more appropriate for this one?

Comment (jfehrle) [reply doc/sphinx/language/extensions/evars.rst:19]: And a similar change unsolvable->unsolved a couple lines below.

**Review by jfehrle — COMMENTED** [review]
Just a few more tweaks if you don't mind, then I think it's ready to go.

File: doc/sphinx/language/extensions/evars.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (Mbodin) [reply doc/sphinx/language/extensions/evars.rst:49]: I know this usage, yeah (although it tends to make it slightly more difficult to read for non-natives… that happen to constitute the majority of the target audience of this documentation ☹). My point of confusion is mostly due to the sentence as a whole here: there are three occurrences of the word `variable`… referring to three slightly different things! I would like to try rephrasing it. Let me try again ☺

[the remark this answers is not in the document]
Comment (Mbodin) [reply doc/sphinx/language/extensions/evars.rst]: I tried to rephrase the sentence in the last commit. What do you think?

File: doc/sphinx/language/extensions/evars.rst [hunk 8]
```diff
@@ -16,10 +16,10 @@ Existential variables
...
 
-Existential variables are generated in place of unsolvable implicit
+Existential variables are generated in place of unsolved implicit
 arguments or “_” placeholders when using commands such as ``Check`` (see
 Section :ref:`requests-to-the-environment`) or when using tactics such as
-:tacn:`refine`, as well as in place of unsolvable instances when using
+:tacn:`refine`, as well as in place of unsolved instances when using
 tactics such that :tacn:`eapply`. An existential
 variable is defined in a context, which is the context of variables of
 the placeholder which generated the existential variable, and a type,
```
Comment (jfehrle) [inline doc/sphinx/language/extensions/evars.rst:25]: Looks good.

It's hard to read this section without finding more text to improve.  The text from "An existential variable is defined..." through the next paragraph is really poorly worded.  It probably deserves a very short example (maybe inline in the text).
If you want to fix that too, great and if not, I'll merge this as it is.

Comment (Mbodin) [reply doc/sphinx/language/extensions/evars.rst:25]: “It's hard to read this section without finding more text to improve.” → I fully agree! I started on this because I noticed a typo… and then I started to understand that I should probably also rephrase a sentence that was really difficult to read, and it keeps going 😅​

To be honest, I don’t think that I’m the most relevant person to improve these sentences: if you don’t mind me to pause on this PR for now, I would prefer to. Thanks for the comments! ☺

Comment (jfehrle) [reply doc/sphinx/language/extensions/evars.rst:25]: I will merge shortly then.

> I don’t think that I’m the most relevant person to improve these sentence

If you're interested in improving wording or the doc in general, don't let that stop you.  It works best with a knowledgeable reviewer.  I've been slowly improving the doc over the last year even though there are many parts of Coq, some basic, that I didn't understand at the outset.

In many places it's hard to figure out what the text means.  Updating wording so it's clear is progress, even if the clear statement is not 100% correct, because the added clarity makes it possible for readers to identify the errors.

**Comment by jfehrle** [comment]
Ah, would you mind squashing before I merge?

**Comment by Mbodin** [comment]
I tried to squash, but it seems that it didn’t work as expected. I kept a local copy. I’m going to retry.

**Comment by Mbodin** [comment]
It seems better squashed like that ☺

**Comment by Mbodin** [comment]
I just understood what happened: in my first attempt, I did `git rebase -i HEAD~10`, only changing my commits expect the first one to `squash`… but by doing so, it somehow repushed the other unrelated commits. Sorry for the noise 😕​

**Comment by Zimmi48** [comment]
No worries, these things happen.

**Review by jfehrle — APPROVED** [review]
Looks good, thanks for your contribution!

**Comment by jfehrle** [comment]
@coqbot: merge now
